setup.py

```diff
@@ -270,133 +246,52 @@ def build_tvtk_classes_zip():
     gen_tvtk_classes_zip()
-    sys.path.remove(tvtk_dir)
-    sys.path.remove(MY_DIR)
-
-
-class MyBuild(build.build):
-    """ A build hook to generate the documentation.
-
-        We sub-class numpy.distutils' build command because we're relying on
-        numpy.distutils' setup method to build python extensions.
-
-    """
-
-    def run(self):
-        build_tvtk_classes_zip()
-        build.build.run(self)
 
 
-class MyBuildSrc(my_build_src_super):
-    """Build hook to generate the TVTK ZIP files.
-
-    We do it here also because for editable installs, setup.py build is not
-    called.
-    """
+class MyBuildPy(build_py):
+    """A build hook to generate the documentation."""
 
     def run(self):
         build_tvtk_classes_zip()
-        build_src_run(self)
-
+        super().run()
 
-class MyDevelop(develop.develop):
-    """ A hook to build the TVTK ZIP file on develop.
 
-        Subclassing setuptools' command because numpy.distutils doesn't
-        have an implementation.
-
-    """
+class MyDevelop(develop):
+    """A hook to build the TVTK ZIP file on develop."""
 
     def run(self):
         # Make sure that the 'build_src' command will
         # always be inplace when we do a 'develop'.
-        self.reinitialize_command('build_src', inplace=1)
+        self.reinitialize_command('build_ext', inplace=1)
 
         # tvtk_classes.zip always need to be created on 'develop'.
         build_tvtk_classes_zip()
 
-        develop.develop.run(self)
-
-
-class MyInstallData(install_data.install_data):
-    """ An install hook to copy the generated documentation.
-
-        We subclass numpy.distutils' command because we're relying on
-        numpy.distutils' setup method to build python extensions.
-
-    """
-
-    def run(self):
-        install_data_command = self.get_finalized_command('install_data')
-        for project in list_doc_projects():
-            install_data_command.data_files.extend(
-                                    list_docs_data_files(project))
-
-        # make sure tvtk_classes.zip always get created before putting it
-        # in the install data.
-        build_tvtk_classes_zip()
-        tvtk_dir = 'tvtk'
-        install_data_command.data_files.append(
-            (tvtk_dir, [join(tvtk_dir, 'tvtk_classes.zip')]))
-
-        install_data.install_data.run(self)
-
-
-class MyClean(clean.clean):
-    """Reimplements to remove the extension module array_ext to guarantee a
-    fresh rebuild every time. The module hanging around could introduce
-    problems when doing develop for a different vtk version."""
-    def run(self):
-        MY_DIR = os.path.dirname(__file__)
-
-        ext_file = os.path.join(
-            MY_DIR,
-            "tvtk",
-            "array_ext" + (".pyd" if sys.platform == "win32" else ".so")
-        )
-
-        if os.path.exists(ext_file):
-            print("Removing in-place array extensions {}".format(ext_file))
-            os.unlink(ext_file)
```

Comment:
This wasn't working anyway, on my system the generated file is `tvtk/array_ext.cpython-311-x86_64-linux-gnu.so` for example. We can revisit at some point if needed